src/consumeroffset/kafka_offset_collect.go

```diff
@@ -74,39 +75,70 @@ func collectOffsetsForConsumerGroup(
 						clientPartitionWg.Done()
 						consumerGroupPartitionWg.Done()
 					}()
+					var partitionLag partitionLagResult
```

Comment:
Is there any particular reason to define this variable and modify it in `collectClientPartitionOffsetMetrics`? Another possibility would be `collectClientPartitionOffsetMetrics` returning a `partitionLagResult`

Reply:
This is more similar to the pattern normally used in Go for readers/marshalers, and also, since collectClientPartitionOffsetMetrics is calling return a couple times to interrupt the execution, having a retuned struct was making me return an empty struct, making the code less clear.

Reply:
I usually find the side-effects approach more confusing (unless there are only a couple of arguments and the naming is crystal clear, like go unmarshalers). Regarding the return used to interrupt the execution, the current approach has the same problem partially modifying the argument, hasn't it?

Anyway, I am not against it I just wanted to know if there was particular reasons to choose this approach. Thank you for clarifying!